java/arcs/core/util/UnionFind.kt

```diff
@@ -13,18 +13,30 @@ package arcs.core.util
 
 /**
  * A union-find datastructure for computing equivalence classes consisting of [E] instances.
- * This datastructure is not thread safe.
+ * Optionally, a value of type [I] may be associated with the equivalence class. If the client
+ * does not need to associated additional information with the class, they may use the [Unit]
+ * type for [I]. This datastructure is not thread safe.
  */
-class UnionFind<E> {
+class UnionFind<E, I> {
     /**
-     * Unifies the equivalence classes of elements [e1] and [e2]. If either
-     * element is not present in any set, it is added.
+     * Unifies the equivalence classes of elements [e1] and [e2]. If either element is not present
```

Comment:
Just like the java style guidelines, it would be best to have the active sentence fragment on its own line, with a blank line between it and the rest of the details.

```kotlin
/**
 * Unifies the equivalence classes of elements [e1] and [e2].
 *
 * If either element is not present .....
 */
```